langchain/chains/summarize/__init__.py

```diff
@@ -49,16 +50,22 @@ def _load_map_reduce_chain(
     collapse_llm: Optional[BaseLanguageModel] = None,
     verbose: Optional[bool] = None,
     token_max: int = 3000,
+    callbacks: Callbacks = None,
```

Comment:
this introduces the converse problem, of making it hard to create a map reduce docs chain with *only* callbacks on the MapReduceDocumentsChain.

maybe we call this arg something like `shared_callbacks`? and for the final map reduce chain we check if kwargs contains 'callbacks', and only used shared_callbacks if it doesn't

Reply:
is that kind of flexibility thing that people want? I think this is an area where I'm a little confused on best practices in langchain. 

What I feel like I'm seeing in *most* of the codebase is that callbacks are passed into agent/chain/llm constructors or utility functions, and that means "all children of this object get the same callbacks" are passed down to child chains / llms / etc.